tests/utils.py

```diff
@@ -328,17 +333,14 @@ def cleanup():
                 cleanup_func(cleanup)
 
     else:
-        hs = homeserverToUse(
-            name,
-            datastore=datastore,
-            config=config,
-            version_string="Synapse/tests",
-            tls_server_context_factory=Mock(),
-            tls_client_options_factory=Mock(),
-            reactor=reactor,
-            **kargs
+        hs = homeserver_to_use(
+            name, config=config, version_string="Synapse/tests", reactor=reactor,
         )
 
+        inject_cache(hs, kwargs)
+
+        hs.datastore = datastore
```

Comment:
`datastore=datastore` was added previously, but `get_datastore()` is not annotated with `@cache_in_self`, so i'm including this here for any fucky sideeffects it might have